pkg/scheduler/plugins/binpack/binpack.go

```diff
@@ -0,0 +1,230 @@
+		length++
+	} else {
+		length += extendLength
+	}
+	msg := make([]string, 0, length)
+	msg = append(msg,
+		fmt.Sprintf(resourceFmt, BinpackWeight, w.BinPackingWeight),
+		fmt.Sprintf(resourceFmt, BinpackCPU, w.BinPackingCPU),
+		fmt.Sprintf(resourceFmt, BinpackMemory, w.BinPackingMemory),
+	)
+
+	if len(w.BinPackingResources) == 0 {
+		msg = append(msg, "no extend resources.")
+	} else {
+		for name, weight := range w.BinPackingResources {
+			msg = append(msg, fmt.Sprintf(resourceFmt, name, weight))
+		}
+	}
+	return strings.Join(msg, ", ")
+}
+
+type binpackPlugin struct {
+	// Arguments given for the plugin
+	weight priorityWeight
+}
+
+//New function returns prioritizePlugin object
+func New(aruguments framework.Arguments) framework.Plugin {
+	weight := calculateWeight(aruguments)
+	return &binpackPlugin{weight: weight}
+}
+
+func calculateWeight(args framework.Arguments) priorityWeight {
+	/*
+	   User Should give priorityWeight in this format(binpack.weight, binpack.cpu, binpack.memory).
+	   Support change the weight about cpu, memory and additional resource by arguments.
+
+	   actions: "enqueue, reclaim, allocate, backfill, preempt"
+	   tiers:
+	   - plugins:
+	     - name: binpack
+	       arguments:
+	         binpack.weight: 10
+	         binpack.cpu: 5
+	         binpack.memory: 1
+	         binpack.resources: nvidia.com/gpu, example.com/foo
+	         binpack.resources.nvidia.com/gpu: 2
+	         binpack.resources.example.com/foo: 3
+	*/
+	// Values are initialized to 1.
+	weight := priorityWeight{
+		BinPackingWeight:    1,
+		BinPackingCPU:       1,
+		BinPackingMemory:    1,
+		BinPackingResources: make(map[v1.ResourceName]int),
+	}
+
+	// Checks whether binpack.weight is provided or not, if given, modifies the value in weight struct.
+	args.GetInt(&weight.BinPackingWeight, BinpackWeight)
+	// Checks whether binpack.cpu is provided or not, if given, modifies the value in weight struct.
+	args.GetInt(&weight.BinPackingCPU, BinpackCPU)
+	if weight.BinPackingCPU < 0 {
+		weight.BinPackingCPU = 1
+	}
+	// Checks whether binpack.memory is provided or not, if given, modifies the value in weight struct.
+	args.GetInt(&weight.BinPackingMemory, BinpackMemory)
+	if weight.BinPackingMemory < 0 {
+		weight.BinPackingMemory = 1
+	}
+
+	resourcesStr := args[BinpackResources]
+	resources := strings.Split(resourcesStr, ",")
+	for _, resource := range resources {
+		resource = strings.TrimSpace(resource)
+		if resource == "" {
+			continue
+		}
+
+		// binpack.resources.[ResourceName]
+		resourceKey := BinpackResourcesPrefix + resource
+		resourceWeight := 1
+		args.GetInt(&resourceWeight, resourceKey)
+		if resourceWeight < 0 {
+			resourceWeight = 1
+		}
+		weight.BinPackingResources[v1.ResourceName(resource)] = resourceWeight
+	}
+
+	return weight
+}
+
+func (bp *binpackPlugin) Name() string {
+	return PluginName
+}
+
+func (bp *binpackPlugin) OnSessionOpen(ssn *framework.Session) {
+	glog.V(3).Infof("Enter binpack plugin ...")
```

Comment:
Other plugins do not log these info. Maybe we could use `glog.V(4)` instead of `glog.V(3)` if needed.

Reply:
use glog.V(4)